### DISCUSSION TITLE
[off-topic] Why do so many people install harfbuzz?

### DISCUSSION THREAD

**Discussion by vegerot** [harfbuzz/harfbuzz#4031] [Q&A] [answered]
harfbuzz is a popular library, but I was still surprised to see it at the top of [Homebrew's most installed formulae](https://formulae.brew.sh/analytics/install-on-request/365d/).  To clarify, this does not include dependencies of formulae (you can find that [here](https://formulae.brew.sh/analytics/install/365d/)) but only specifically running `brew install harfbuzz`.

I was talking to a friend about this yesterday and we were both surprised.  One hypothesis is that the number of "manual" installs is inflated by CI jobs.

What do you think?

**Comment by behdad** [comment]
I'm surprised by this as well.

**Comment by alerque** [comment, marked as answer]
I suspect the CI hypothesis is at least partially correct. I've observed something similar in other ecosystems. A dramatic one is Lua: if you look at the install counts on luarocks.org many packages are just wildly more installed than others, and the difference does not track even in the same order or magnitude as GitHub stars or other metrics of interest. Having talked to people that run the servers passing out manifests they say the difference is ones that get pulled explicitly by CI runs on active projects.

Another thing to consider is that the stats might not be as accurate as you think as far as differentiating explicit installations vs. dependencies. A know for a fact on the 3 distros I've done in depth dives on (PLD, Nix, Arch) that the deps vs. explicit install markers make sense in isolation but are very easily messed up and skewed by any number of factors. I'm not as intimately familiar with Homebrew but I suspect is is susceptible to similar skewing.

Because harfbuzz is a dependency of so many other things it's probably not infrequent that developers end up installing it manually at *some* point even if they only did so as a build dependency for some other project (or some script did it for them).

**Reply by vegerot** [reply]
Agreed.  It would be cool to find where this happens
